engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java

```diff
@@ -303,7 +303,15 @@ protected boolean equals(int errorCode, String sqlState) {
   }
 
   public static boolean checkDeadlockException(SQLException sqlException) {
-    String sqlState = sqlException.getSQLState().toUpperCase();
+    String sqlState = sqlException.getSQLState();
+    if (sqlState != null) {
+      sqlState = sqlState.toUpperCase();
+      
```

Comment:
Let's avoid the empty lines
```suggestion
      sqlState = sqlState.toUpperCase();
```